src/include/replayer.h

```diff
@@ -0,0 +1,85 @@
+#include <fstream>
+#include <vector>
+#include <mutex>
+#include <chrono>
+#include "debug.h"
+//#include "nccl.h"?
+
+// API opcode covered by rccl replayer
+typedef enum {
+  rrBroadcast = 0,
+  rrReduce = 1,
+  rrAllGather = 2,
+  rrReduceScatter = 3,
+  rrAllReduce = 4,
+  rrSend = 6,
+  rrRecv = 7,
+  rrAllToAll = 8,
+  rrAllToAllv = 9 ,
+  rrGather = 5,
+  rrScatter = 13,
+  rrGroupStart = 10,
+  rrGroupEnd = 11,
+  rrCommInitRank = 20,
+  rrCommSplit = 24,
+  rrCommFinalize = 25,
+  rrCommDestroy = 26,
+  rrCommAbort = 27,
+  rrCommRegister = 30,
+  rrCommDeregister = 31
+} rcclCall_t;
+
+using namespace std::chrono;
+struct CollectiveCall {
```

Comment:
We should rename this to be more clear that this is a rccl addition.
It might not even always be a collective call.

rcclReplayCall?